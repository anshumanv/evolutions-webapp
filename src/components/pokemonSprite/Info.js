import React from 'react';
import PropTypes from 'prop-types';
import Stats from './Stats';
import '../../css/PokemonSprite.css';

const Info = ({ info }) => (
  <div className="PokemonSprite-container">
    <div> {`[#${info.num}] ${info.species} (${info.types.join(', ')})`} </div>
    <div>
      Weight: {info.weightkg} kg, Height: {info.heightm} m
    </div>
    <div> Abilities: {Object.values(info.abilities).join(', ')} </div>
    <Stats baseStats={info.baseStats} />
  </div>
);

Info.propTypes = {
  info: PropTypes.object,
};

export default Info;
