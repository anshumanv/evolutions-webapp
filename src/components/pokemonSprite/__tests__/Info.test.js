import React from 'react';
import { shallow } from 'enzyme';
import Info from '../Info';

describe('Info component', () => {
  it('renders', () => {
    const sampleData = {
      species: 'pikachu',
      num: 25,
      types: ['Electric'],
      weight: 6,
      heightm: 0.4,
      abilities: ['Static', 'Lightning Rod'],
    };
    const component = shallow(<Info info={sampleData} />);
    expect(component).toMatchSnapshot();
  });
});
