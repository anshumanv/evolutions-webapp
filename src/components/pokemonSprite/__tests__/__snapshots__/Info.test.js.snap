// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info component renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Info
    info={
        Object {
            "abilities": Array [
              "Static",
              "Lightning Rod",
            ],
            "heightm": 0.4,
            "num": 25,
            "species": "pikachu",
            "types": Array [
              "Electric",
            ],
            "weight": 6,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
           
          pikachu [#25] (Electric)
           
</div>,
        <div>
          Weight: 
           kg, Height: 
          0.4
           m
</div>,
        <div>
           Abilities: 
          Static, Lightning Rod
           
</div>,
        <Stats
          baseStats={undefined}
/>,
      ],
      "className": "PokemonSprite-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "pikachu [#25] (Electric)",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          "pikachu [#25] (Electric)",
          " ",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Weight: ",
            undefined,
            " kg, Height: ",
            0.4,
            " m",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Weight: ",
          undefined,
          " kg, Height: ",
          0.4,
          " m",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " Abilities: ",
            "Static, Lightning Rod",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " Abilities: ",
          "Static, Lightning Rod",
          " ",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "baseStats": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
             
            pikachu [#25] (Electric)
             
</div>,
          <div>
            Weight: 
             kg, Height: 
            0.4
             m
</div>,
          <div>
             Abilities: 
            Static, Lightning Rod
             
</div>,
          <Stats
            baseStats={undefined}
/>,
        ],
        "className": "PokemonSprite-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "pikachu [#25] (Electric)",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            "pikachu [#25] (Electric)",
            " ",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Weight: ",
              undefined,
              " kg, Height: ",
              0.4,
              " m",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Weight: ",
            undefined,
            " kg, Height: ",
            0.4,
            " m",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " Abilities: ",
              "Static, Lightning Rod",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " Abilities: ",
            "Static, Lightning Rod",
            " ",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "baseStats": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
